analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,214 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path.
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm).
+    The Impact parameter significance is stored in the particle's path.
+    
+    New attributes for particles.path (from compute_IP) :
+    *   closest_t = time of closest approach to the primary vertex.
+    *   IP = signed impact parameter
+    *   IPx, IPy, IPz = coordinates of the point of closest approach to the
+        primary vertex, with respets to the primary vertex
+        
+    New attributes for particles.path (from compute_theta_0) :
+    *   theta_0 = 1/sqrt(2) * gaussian width of the scattering angle
+        due to the beam pipe
+    *   xX_0 = distance in number of X_0 the radiation length the particles goes through
+    
+    New attributes for particles.path (from compute_IP_signif) :
+    *   IP_signif = the impact parameter significance. To get the uncertainty, just compute IP/IP_signif
```

Comment:
please store both the uncertainty and the significance. 
getting the uncertainty in this way feels weird